tests/test_parser_json_snippets.py

```
@@ -0,0 +1,333 @@
+def test_section_and_subsection(parser):
+    got = parser.parse("Intro\n== First ==\nHello\n=== Sub ===\nWorld")
+    sec = got["children"][1]
+    assert sec["type"] == "section" and sec["properties"]["level"] == 2
+    sub = sec["children"][1]
+    assert sub["type"] == "section" and sub["properties"]["level"] == 3
+
+def test_full_boston_snapshot(parser):
+    here = os.path.dirname(__file__)
+    inp = os.path.join(here, "fixtures", "boston_input.txt")
+    out = os.path.join(here, "fixtures", "boston_output.json")
+    wikicode = open(inp, encoding="utf-8").read()
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)